test/integration/fdp-class.spec.ts

```diff
@@ -460,5 +460,88 @@ describe('Fair Data Protocol class', () => {
       expect(sharedData.meta).toBeDefined()
       expect(sharedData.source_address).toHaveLength(40)
     })
+
+    it('should receive information about shared file', async () => {
+      const fdp = createFdp()
+      const user = generateUser(fdp)
+      const pod = generateRandomHexString()
+      const fileSizeSmall = 100
+      const contentSmall = generateRandomHexString(fileSizeSmall)
+      const filenameSmall = generateRandomHexString() + '.txt'
+      const fullFilenameSmallPath = '/' + filenameSmall
+      await topUpAddress(fdp)
+
+      await fdp.account.register(user.username, user.password)
+      await fdp.personalStorage.create(pod)
+      await fdp.file.uploadData(pod, fullFilenameSmallPath, contentSmall)
+
+      const sharedReference = await fdp.file.share(pod, fullFilenameSmallPath)
+      const sharedData = await fdp.file.getSharedInfo(sharedReference)
+
+      expect(sharedData.meta).toBeDefined()
+      expect(sharedData.meta.pod_name).toEqual(pod)
+      expect(sharedData.meta.file_path).toEqual('/')
+      expect(sharedData.meta.file_name).toEqual(filenameSmall)
+      expect(sharedData.meta.file_size).toEqual(fileSizeSmall)
+      expect(sharedData.source_address).toHaveLength(40)
+    })
+
+    it('should save shared file to a pod', async () => {
+      const fdp = createFdp()
+      const fdp1 = createFdp()
+      const user = generateUser(fdp)
+      const user1 = generateUser(fdp1)
+      const pod = generateRandomHexString()
+      const pod1 = generateRandomHexString()
+      const fileSizeSmall = 100
+      const contentSmall = generateRandomHexString(fileSizeSmall)
+      const filenameSmall = generateRandomHexString() + '.txt'
+      const fullFilenameSmallPath = '/' + filenameSmall
+      await topUpAddress(fdp)
+      await topUpAddress(fdp1)
+
+      await fdp.account.register(user.username, user.password)
```

Comment:
is this registering on blockchain still necessary?

Reply:
I think no. This PR was created before we raised the issue of moving away from permanent registrations on the blockchain. In the next PR I will leave registration only in specific tests